### PR TITLE
Improve glusterfs backend

### PR DESCRIPTION
**Pull request by arogge**
* add logging options to glusterfs backend
* add a systemtest to test the glusterfs backend

### CODE REVIEW

File: core/src/stored/backends/gfapi_device.cc [hunk 1]
```diff
@@ -338,20 +342,23 @@ int gfapi_device::d_open(const char* pathname, int flags, int mode)
...
             default:
+              Mmsg1(errmsg, _("Unable to parse device option: %s\n"), bp);
+              Emsg0(M_FATAL, 0, errmsg);
+              goto bail_out;
               break;
           }
+          done = true;
         }
       }
-
```
Comment (pstorz) [inline core/src/stored/backends/gfapi_device.cc]: Was this block intentionally removed?

Comment (arogge) [reply core/src/stored/backends/gfapi_device.cc]: If you look closely, you'll find it in the `switch()`'s default-case

File: core/src/stored/backends/gfapi_device.cc [hunk 2]
```diff
@@ -338,20 +342,23 @@ int gfapi_device::d_open(const char* pathname, int flags, int mode)
           switch (device_options[i].type) {
             case argument_uri:
               gfapi_uri_ = bp + device_options[i].compare_size;
-              done = true;
+              break;
+            case argument_logfile:
+              gfapi_logfile_ = bp + device_options[i].compare_size;
+              break;
+            case argument_loglevel:
+              gfapi_loglevel_ =
```
Comment (franku) [inline core/src/stored/backends/gfapi_device.cc]: In case of error  gfapi_loglevel_ will be zero.

File: core/src/stored/backends/gfapi_device.h [hunk 3]
```diff
@@ -40,6 +40,8 @@ class gfapi_device : public Device {
  private:
   char* gfapi_configstring_;
   char* gfapi_uri_;
+  char* gfapi_logfile_;
```
Comment (franku) [inline core/src/stored/backends/gfapi_device.h]:
```suggestion
  char* gfapi_logfile_{};
```

Comment (franku) [reply core/src/stored/backends/gfapi_device.h]: Should double check if the constructor is really called. Ohterwise this has no effect.

File: core/src/stored/backends/gfapi_device.cc (hunk 2 above)
Comment (arogge) [reply core/src/stored/backends/gfapi_device.cc]: this is expected and not a problem.

**Pull request merged**